packages/agent-utils/src/agent/agent-function.ts

```diff
@@ -6,25 +6,63 @@ import {
 } from "./prompts";
 
 import { Result, ResultErr, ResultOk } from "@polywrap/result";
-import { ChatCompletionFunctions } from "openai";
+import { ChatCompletionFunctions, ChatCompletionRequestMessage } from "openai";
 import JSON5 from "json5";
+import { ChatRole } from "../llm";
 
-export interface AgentChatMessage {
+export interface AgentOutputMessage {
```

Comment:
This file is getting a bit crowded, maybe we should structure it like so:
```
agent/
  agent-function/
    index.ts
    execute.ts
    messages.ts
```

Reply:
Side note: we have a few different naming conventions going on, would be happy to pick one and stick with it.

Reply:
Yep, will refactor to external files
Got any thoughts on namings?

Reply:
Did some renaming/refactor, check it out now

Reply:
Looks great!